synse/http_integration_test.go

```diff
@@ -0,0 +1,473 @@
+}
+
+func TestIntegration_Version(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+
+	client, err := NewHTTPClientV3(&Options{
+		Address: "localhost:5000",
+	})
+	assert.NotNil(t, client)
+	assert.NoError(t, err)
+
+	version, err := client.Version()
+	assert.NoError(t, err)
+	assert.Equal(t, "3.0.0-alpha.3", version.Version)
+	assert.Equal(t, "v3", version.APIVersion)
+}
+
+func TestIntegration_Config(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+
+	client, err := NewHTTPClientV3(&Options{
+		Address: "localhost:5000",
+	})
+	assert.NotNil(t, client)
+	assert.NoError(t, err)
+
+	config, err := client.Config()
+	assert.NoError(t, err)
+	assert.Equal(t, "debug", config.Logging)
+	assert.True(t, config.PrettyJSON)
+	assert.Equal(t, "en_US", config.Locale)
+	assert.Equal(t, 1, len(config.Plugin.TCP))
+	assert.Equal(t, "emulator:5001", config.Plugin.TCP[0])
+	assert.Equal(t, 0, len(config.Plugin.Unix))
+	// assert.Equal(t, 180, config.Cache.Device.RebuildEvery) // TODO - update scheme
+	assert.Equal(t, 300, config.Cache.Transaction.TTL)
+	assert.Equal(t, 3, config.GRPC.Timeout)
+	assert.False(t, config.Metrics.Enabled)
+}
+
+func TestIntegration_Plugins(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+
+	client, err := NewHTTPClientV3(&Options{
+		Address: "localhost:5000",
+	})
+	assert.NotNil(t, client)
+	assert.NoError(t, err)
+
+	plugins, err := client.Plugins()
+	assert.NoError(t, err)
+	assert.Equal(t, 1, len(plugins))
+
+	plugin := plugins[0]
+	assert.Equal(t, "emulator plugin", plugin.Name)
+	assert.Equal(t, "vaporio", plugin.Maintainer)
+	assert.Equal(t, "vaporio/emulator-plugin", plugin.Tag)
+	assert.Equal(t, "A plugin with emulated devices and data", plugin.Description)
+	assert.Equal(t, "4032ffbe-80db-5aa5-b794-f35c88dff85c", plugin.ID)
+	assert.True(t, plugin.Active)
+}
+
+func TestIntegration_PluginInfo(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+
+	client, err := NewHTTPClientV3(&Options{
+		Address: "localhost:5000",
+	})
+	assert.NotNil(t, client)
+	assert.NoError(t, err)
+
+	plugin, err := client.Plugin("4032ffbe-80db-5aa5-b794-f35c88dff85c")
+	assert.NoError(t, err)
+	assert.Equal(t, "emulator plugin", plugin.Name)
+	assert.Equal(t, "vaporio", plugin.Maintainer)
+	assert.Equal(t, "vaporio/emulator-plugin", plugin.Tag)
+	assert.Equal(t, "A plugin with emulated devices and data", plugin.Description)
+	assert.Equal(t, "4032ffbe-80db-5aa5-b794-f35c88dff85c", plugin.ID)
+	assert.True(t, plugin.Active)
+
+	assert.Equal(t, "emulator:5001", plugin.Network.Address)
+	assert.Equal(t, "tcp", plugin.Network.Protocol)
+	assert.Equal(t, "3.0.0-alpha.3", plugin.Version.PluginVersion)
+	assert.Equal(t, "3.0.0-alpha.1", plugin.Version.SDKVersion)
+	assert.NotEmpty(t, plugin.Version.BuildDate)
+	assert.Equal(t, "4234777", plugin.Version.GitCommit)
+	assert.Equal(t, "3.0.0-alpha.3", plugin.Version.GitTag)
```

Comment:
the git tag / commit can just be an empty check since this will change depending on what the `:latest` image is